tracer/test/Datadog.Trace.Tests/Configuration/TracerSettingsTests.cs

```diff
@@ -194,6 +198,644 @@ public void SetServerHttpCodes()
+            var settings = new TracerSettings(source);
+
+            settings.IsDataStreamsMonitoringEnabled.Should().Be(expected);
+        }
+
+        [Theory]
+        [MemberData(nameof(BooleanTestCases), false)]
+        public void IsRareSamplerEnabled(string value, bool expected)
+        {
+            var source = CreateConfigurationSource((ConfigurationKeys.RareSamplerEnabled, value));
+            var settings = new TracerSettings(source);
+
+            settings.IsRareSamplerEnabled.Should().Be(expected);
+        }
+
+        [Theory]
+        [MemberData(nameof(BooleanTestCases), false)]
+        public void IsRunningInAzureAppService(string value, bool expected)
+        {
+            var source = CreateConfigurationSource((ConfigurationKeys.RareSamplerEnabled, value));
+            var settings = new TracerSettings(source);
+
+            settings.IsRareSamplerEnabled.Should().Be(expected);
+        }
+
+        [Fact]
+        public void DisableTracerIfNoApiKeyInAas()
+        {
+            var source = CreateConfigurationSource((ConfigurationKeys.AzureAppService.AzureAppServicesContextKey, "1"));
+            var settings = new TracerSettings(source);
+
+            settings.TraceEnabled.Should().BeFalse();
+        }
+
+        [Theory]
+        [InlineData("test1,, ,test2", false, false, new[] { "TEST1", "TEST2" })]
+        [InlineData("test1,, ,test2", true, true, new[] { "TEST1", "TEST2" })]
+        [InlineData(null, true, true, new[] { "azuredefault" })]
+        [InlineData(null, false, true, new[] { "SERVERLESS" })]
+        [InlineData(null, false, false, new string[0])]
+        [InlineData("", true, true, new string[0])]
+        public void HttpClientExcludedUrlSubstrings(string value, bool isRunningInAppService, bool isRunningInLambda, string[] expected)
+        {
+            if (expected.Length == 1 && expected[0] == "azuredefault")
+            {
+                expected = ImmutableAzureAppServiceSettings.DefaultHttpClientExclusions.Split(',').Select(s => s.Trim()).ToArray();
+            }
+
+            var oldMetadata = Serverless.Metadata;
+
+            try
+            {
+                var source = CreateConfigurationSource(
+                    (ConfigurationKeys.HttpClientExcludedUrlSubstrings, value),
+                    (ConfigurationKeys.AzureAppService.AzureAppServicesContextKey, isRunningInAppService ? "1" : "0"));
+
+                if (isRunningInLambda)
+                {
+                    var metadata = Serverless.LambdaMetadata.CreateForTests(true, "functionName", "handlerName", "serviceName", "serverless");
+                    Serverless.SetMetadataTestsOnly(metadata);
+                }
+
+                var settings = new TracerSettings(source);
+
+                settings.HttpClientExcludedUrlSubstrings.Should().BeEquivalentTo(expected);
+            }
+            finally
+            {
+                Serverless.SetMetadataTestsOnly(oldMetadata);
```

Comment:
Doesn't look like we're using this in any other tests yet, but I wonder if it's worth being safe and adding this test to a "serverless" collection so that we don't run it in parallel with other serverless tests? 

On second thoughts, not necessary, just remember when it comes up in test rotation 😉 

Alternatively, maybe we should update `ServerlessMetadata` to be more like `ImmutableAzureAppServicesSettings` (i.e. built from the provided `IConfigurationSource`) so it's not an issue 🤔  Not for this PR, just thinking out loud